sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/UseUriInsteadOfString.cs

```diff
@@ -106,6 +112,33 @@ protected sealed override void Initialize(SonarAnalysisContext context)
                     }
                 },
                 SyntaxKind.PropertyDeclaration);
+
+            context.RegisterSyntaxNodeActionInNonGenerated(
+                c =>
+                {
+                    var invokedMethod = c.SemanticModel.GetSymbolInfo(c.Node).Symbol as IMethodSymbol;
+                    if (invokedMethod == null)
+                    {
+                        return;
+                    }
+
+                    if (invokedMethod.IsInType(KnownType.System_Uri))
```

Comment:
Merge this if condition with the previous one. Regarding your question, I think you did well and I don't think we need to update RSPEC (that make sense not to raise on URI methods, isn't it?).